### PR TITLE
Full support for Dropdown alignment options

### PR DESCRIPTION
**Pull request by vepanimas**
There are a lot of issues related to incomplete dropdown alignment support. In this PR I added support for all PopperJS placement options.

Two new classes were introduced: `.dropdown-menu-center` (for all directions) and `.dropdown-menu-bottom` (only for `dropleft` and `dropright`).

I am open to discussion, feel free to comment.

### CODE REVIEW

**Review by Johann-S — CHANGES_REQUESTED** [review]
Please add unit tests

**Comment by vepanimas** [comment]
Thx for the review! Here you can see an example https://r4j6xr2x0m.codesandbox.io/, the source code is here https://codesandbox.io/s/r4j6xr2x0m. And of course I'll add unit tests, I just need some time.

**Comment by vepanimas** [comment]
@Johann-S, done!

**Review by Johann-S — APPROVED** [review]
LGTM ! Thanks @vepanimas !

Before merging that I need a review from @andresgalante  or @mdo about your SCSS code 👍 

And if it's fine I think we'll merge that in our next major release (4.2)

BTW it'll partially fixed #22962

File: site/docs/4.1/components/dropdowns.md
Comment (andresgalante) [inline site/docs/4.1/components/dropdowns.md]: @mdo This is a bit hard to understand, I'd rewrite it as:

> Apply `.dropdown-menu-right` or `.dropdown-menu-center` as a modifier to `.dropdown-menu` to right or center align the menu on a dropdown or dropup. And apply `.dropdown-menu-bottom ` or `.dropdown-menu-center` as a modifier to `.dropdown-menu` to bottom or center align the menu on a dropright or dropleft.

What do you think?

**Review by andresgalante — COMMENTED** [review]
@vepanimas thanks a lot for you contribution! 👍The code is sound 

@Johann-S the position of the menu gets overwritten by popper, the transforms and positions we have in our code base are a nice fallback in case the UI is loaded without JS or popper is not present.

File: site/docs/4.1/components/dropdowns.md
Comment (vepanimas) [reply site/docs/4.1/components/dropdowns.md]: I tried as hard as I could, but I'm not a native speaker =)

Comment (andresgalante) [reply site/docs/4.1/components/dropdowns.md]: @vepanimas don't worry, me neither, that's why I am asking @mdo 😝

Comment (Johann-S) [reply site/docs/4.1/components/dropdowns.md]: Maybe @XhmikosR can help us too 😄 or @patrickhlauke

Comment (XhmikosR) [reply site/docs/4.1/components/dropdowns.md]: @mdo: can you chime in here? Otherwise this can be merged AFAIK.

**Comment by XhmikosR** [comment]
Actually unit tests fail here.

**Comment by XhmikosR** [comment]
Might be something with the conflicts, @Johann-S have a look when you can :)

**Comment by Johann-S** [comment]
Maybe @vepanimas can fix the conflict ?

**Comment by vepanimas** [comment]
@Johann-S , I'll look into it tomorrow.

**Comment by XhmikosR** [comment]
Ping @vepanimas

**Comment by XhmikosR** [comment]
Closing due to lack of response. You can comment here and we can re-open it if you decide to resume working on the requested changes.

**Comment by k-funk** [comment]
I too, would like to see this feature.

**Comment by XhmikosR** [comment]
Not sure this PR can be merged so @k-func you're welcome to submit a new proper PR with unit tests.

**Pull request closed without merging**
